app/javascript/components/workflows/workflow_payload.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Loading, Tabs, Tab,
+} from 'carbon-components-react';
+import AWSSfnGraph from '@tshepomgaga/aws-sfn-graph';
+import { Controlled as CodeMirror } from 'react-codemirror2';
+import '@tshepomgaga/aws-sfn-graph/index.css';
+
+const WorkflowPayload = ({ recordId }) => {
+  const tabLabels = [
+    { name: 'text', text: __('Text') },
+    { name: 'graph', text: __('Graph') },
+  ];
+  const [data, setData] = useState({
+    isLoading: true,
+    payload: '',
+    activeTab: tabLabels[0].name,
+  });
+
+  // TODO: Change the url when the GET automated worflow summary for recordId api is available.
+  useEffect(() => {
+    API.get(`/api/configuration_script_payloads/${recordId}?expand=resources`)
+      .then((response) => {
+        setData({
+          isLoading: false,
+          payload: response.payload,
+        });
+      });
+  }, [recordId]);
+
+  if (data.isLoading) {
+    return (
+      <div className="loadingSpinner">
+        <Loading active small withOverlay={false} className="loading" />
+      </div>
+    );
+  }
+
+  /** Function to set the active tab name */
+  const onTabSelect = (name) => {
+    setData({
+      ...data,
+      activeTab: name,
+    });
+  };
+
+  /** Function to render the code mirror component */
+  const error = false;
+  const renderCodeSnippet = () => (
+    <CodeMirror
+      className={`miq-codemirror workflow-payload ${error ? 'has-error' : ''}`}
+      options={{
+        mode: 'application/ld+json',
+        theme: 'eclipse',
+        lineNumbers: true,
+        lineWrapping: true,
+        foldGutter: true,
+        gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter', 'CodeMirror-lint-markers'],
+        autoCloseBrackets: true,
+        matchBrackets: true,
+        lint: true,
+      }}
+      style={{ height: '300px' }}
+      value={JSON.stringify(JSON.parse(data.payload), null, ' ')}
```

Comment:
Why is the payload parsed then stringified when it already comes as a string?

Reply:
Ohhh so it's to "prettify" the payload.  Interesting.  I wonder if we can prettify it from the API itself.  @agrare ?

Reply:
We could `JSON.pretty_generate()` when we store the payload, I'm thinking it is better to do this when have the context of what the payload is (aka in the provider) versus when we are dealing with these generically in API/UI.  Downside is the payload size in the db will be slightly larger.

It looks like the `YAML` type payloads (AKA ansible playbooks) already have `\n` and pretty-print well

Reply:
Agreed - let's change the backend to pretty_generate that way the front end doesn't need presentation logic other than just knowing the payload_type for syntax highlighting.

Reply:
Merged https://github.com/ManageIQ/manageiq-providers-workflows/pull/19